contracts/on-chain-claim/src/config.rs

```diff
@@ -40,16 +71,47 @@ pub trait ConfigModule {
             return false;
         }
 
-        let address_info = address_info_mapper.get();
+        let address_info = self.get_address_info(address);
+        if address_info.total_epochs_claimed == 0 {
+            return false;
+        }
+
         let current_epoch = self.blockchain().get_block_epoch();
         let missed_epochs = self.get_missed_epochs(current_epoch, address_info.last_epoch_claimed);
 
         missed_epochs > 0 && missed_epochs <= MAX_REPAIR_GAP
     }
 
+    #[view(getCurrentSeason)]
+    fn get_current_season(&self) -> SeasonInfo {
```

Comment:
If you use the VecMapper approach, you could keep SeasonInfo struct (without saving it) and create it on the spot. But at the same time, you could just read the starting epoch, do you checks and just return the season_id. So this function would return a variable of type usize (same type as the VecMapper index).